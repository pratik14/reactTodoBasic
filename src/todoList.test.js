import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import TodoList from './todoList';


it('renders correctly', () => {
  const list = [
    { name: 'Test 1', id: 'xx', completed: false},
    { name: 'Test 2', id: 'yy', completed: false},
  ]
  const tree = renderer
    .create(<TodoList list={list} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('check if pending and completed list are rendering fine', () => {
  const list = [
    { name: 'Test 1', id: 'xx', completed: false},
    { name: 'Test 2', id: 'yy', completed: true},
  ]

  const wrapper = mount(<TodoList list={list} />);
  
  expect(wrapper.find('#pending-tasks').length).toBe(1)
  expect(wrapper.find('#completed-tasks').length).toBe(1)
})
