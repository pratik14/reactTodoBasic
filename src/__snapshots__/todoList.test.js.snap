// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div>
    <h3>
      pending
    </h3>
    <ul
      id="pending-tasks"
    >
      <li>
        <input
          checked={false}
          onClick={[Function]}
          readOnly={true}
          type="checkbox"
        />
        <label>
          Test 1
        </label>
        <button
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          onClick={[Function]}
        >
          Remove
        </button>
      </li>
      <li>
        <input
          checked={false}
          onClick={[Function]}
          readOnly={true}
          type="checkbox"
        />
        <label>
          Test 2
        </label>
        <button
          onClick={[Function]}
        >
          Edit
        </button>
        <button
          onClick={[Function]}
        >
          Remove
        </button>
      </li>
    </ul>
  </div>
</div>
`;
